### ISSUE TITLE
SLF4J Error message Log on first execution.

### ISSUE THREAD

**Issue by feinstein** [awslabs/aws-serverless-java-container#80]
The first time my lambda code runs, it leave a `SLF4J` error message on CloudWatch logs:

![image](https://user-images.githubusercontent.com/8120472/33915841-ac5c9a5c-df8c-11e7-941f-414c6830b3b9.png)

You can see on the above image that I run my lambda code several times and the `SLF4J` error only appears on the first execution.

In order to be sure this was not a problem from my code, I downloaded the `Jersey` sample and uploaded it to my lambda instance. On the first execution there's also an error message, but from `Log4J` this time:

![image](https://user-images.githubusercontent.com/8120472/33915884-13517d72-df8d-11e7-8b70-d9ffa355768b.png)

I am not using `SLF4J` on my project, I am using pure `log4j2` with the AWS Appender. But my code is very simple, just an echo so I can test the whole architecture and even before I placed `log4j2` the `SLF4J` error was still there.

In order to reproduce this issue you have to upload a new `jar` since this only happens on the first execution (cold start) of a lambda function.

If you want I can send you my test code so you can see the `SLF4J` error message yourself.

**Comment by SAPessi** [comment]
Hey @feinstein, are you including Lambda's [log4j appender](http://docs.aws.amazon.com/lambda/latest/dg/java-logging.html) in your dependencies?

**Comment by feinstein** [comment]
Hi @SAPessi, I am, but in any case the second screen shot is the sample code from the library, pure, I didn't modify it, just compiled and uploaded it.

**Comment by SAPessi** [comment]
ok. Let me look into it. The library uses SLF4J as an interface and trusts that an appender is included (the Lambda log4j appender should work). I'll run some tests to figure out what is going on. We'll use this issue to track it.

**Comment by feinstein** [comment]
Great, if you need my project's code, please let me know.

Also, IMHO I believe libraries should default to the no-op SLF4J Appender when none is present, with no error messages, since most applications want libraries to be transparent and not pollute the logs with cryptic error messages.

**Comment by feinstein** [comment]
I am not very much experienced on SLF4J, I did a quick search and this website has the error message and the explanation.

https://www.slf4j.org/manual.html 

So maybe there's a configuration documentation missing, saying that I need to provide a SLF4J binding.

The error message from the sample I believe is due to the lack of Log4j2 Appender configuration, so it will default to the console.

**Comment by SAPessi** [comment]
Hey @feinstein, getting back to this - could you share your pom.xml file?

**Comment by SAPessi** [comment]
The issue happens because there is no dependency in the project that defines an appender for slf4j. I have intentionally not included an appender in the library because I want each developer to use their preferred one. Having said that, I have found a pretty awesome library that defines an appender for slf4j specifically for Lambda - it also includes the request id in the log lines: https://blog.symphonia.io/a-love-letter-to-lambda-logging-974b0eb49273

I have updated all of the samples to include this library.
